OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -679,6 +679,31 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Move and rename Health.Shape to HitShape.Type
+				if (engineVersion < 20170521)
+				{
+					var health = node.Value.Nodes.FirstOrDefault(n => n.Key == "Health");
+					if (health != null)
+					{
+						var shape = health.Value.Nodes.FirstOrDefault(n => n.Key == "Shape");
+						if (shape != null)
+						{
+							RenameNodeKey(shape, "Type");
+							var hitShape = new MiniYamlNode("HitShape", "");
+							hitShape.Value.Nodes.Add(shape);
+							node.Value.Nodes.Add(hitShape);
+							health.Value.Nodes.Remove(shape);
+						}
+						else
+						{
+							var hitShape = new MiniYamlNode("HitShape", "");
+							node.Value.Nodes.Add(hitShape);
+							Console.WriteLine("'Shape' was moved from Health to HitShape and renamed to 'Type'.");
```

Comment:
Do modders need to do any manual work for this?  If not, best to keep it in the changelog instead of logging it tens of times during the upgrade.